### PR TITLE
Fix DTrace profiler integration with CLion 2021.2

### PR DESCRIPTION
**Pull request by ortem**
Fixes https://github.com/intellij-rust/intellij-rust/issues/7727

However, further refactorings on the CLion side should be done to reuse as much code as possible and to avoid such problems in the future.

changelog: Fix DTrace profiler integration with CLion 2021.2

### CODE REVIEW

**Review by Undin — APPROVED** [review]
This change leads to duplication in UI for CLion 2021.1 and inconsistency with C++ UI on 2021.1. Maybe it's worth fixing it?

| Before | After |
| - | - |
|  |  |

**Comment by Undin** [comment]
For QA: this change should be checked with both 2021.1 and 2021.2 major releases

**Comment by ortem** [comment]
@Undin Thanks, I fixed it by adding backcompat code

File: clion/src/main/kotlin/org/rust/clion/profiler/dtrace/RsDTraceProfilerProcess.kt [hunk 1]
```diff
@@ -86,3 +93,5 @@ class RsDTraceProfilerProcess private constructor(
         }
     }
 }
+
+private val BUILD_212: BuildNumber = BuildNumber.fromString("212")!!
```
Comment (Undin) [inline clion/src/main/kotlin/org/rust/clion/profiler/dtrace/RsDTraceProfilerProcess.kt:97]: Let's add `BACKCOMPAT: 2021.1` comment here as well. Just not to forget to drop it

**Review by Undin — APPROVED** [review]

**Comment by ortem** [comment]
bors r=Undin

**Pull request merged**